react/typings/events.d.ts

```diff
@@ -181,6 +181,15 @@ export interface BeginCheckoutData extends EventData {
   currency: string
 }
 
+export interface AddShippingInfoData extends EventData {
+  event: 'addShippingInfo'
+  eventType: 'vtex:addShippingInfo'
+  items?: CartItem[]
+  shippingTier?: string
+  currency: string
+  value?: number
```

Comment:
Three suggestions:

1. Remove `currency` since it's inherited from `EventData`. I've just noticed we also have it in some other recent events.
https://github.com/vtex-apps/google-tag-manager/blob/d6ad9c741ea482e4bbe9dbbadbb7d17da0d47b1a/react/typings/events.d.ts#L169-L197

2. Remove `value` since it shouldn't be part of the input. It's part of the output for the GA4 event format.

3. Remove the optionality of the fields since they should be required.

```suggestion
  items: CartItem[]
  shippingTier: string
```